src/storage/src/hummock/state_store.rs

```diff
@@ -289,6 +283,36 @@ impl HummockStorage {
             .observe(table_counts as f64);
         Ok(None)
     }
+
+    #[allow(clippy::type_complexity)]
+    fn read_filter<R, B>(
+        &self,
+        epoch: HummockEpoch,
+        key_range: &R,
+        vnode_set: Option<&VNodeBitmap>,
+    ) -> HummockResult<(
+        BTreeMap<HummockEpoch, (Vec<SharedBufferBatch>, Vec<UncommittedData>)>,
```

Comment:
Why do we need to return a BTreeMap keyed by epoch instead of a Vec ordered by epoch? I don't see the epoch being used in this PR.

Reply:
Seems unnecessary. I have recovered it to a vec.